backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/EditLabelEventHandler.java

```diff
@@ -88,27 +92,42 @@ public boolean canHandle(IDiagramInput diagramInput) {
     public EventHandlerResponse handle(IEditingContext editingContext, IDiagramContext diagramContext, IDiagramInput diagramInput) {
         this.counter.increment();
 
+        EventHandlerResponse response = null;
         if (diagramInput instanceof EditLabelInput) {
             EditLabelInput input = (EditLabelInput) diagramInput;
+            Diagram diagram = diagramContext.getDiagram();
             if (input.getLabelId().endsWith(LABEL_SUFFIX)) {
-                Diagram diagram = diagramContext.getDiagram();
                 String nodeId = this.extractNodeId(input.getLabelId());
                 var node = this.diagramService.findNodeById(diagram, nodeId);
                 if (node.isPresent()) {
                     this.invokeDirectEditTool(node.get(), editingContext, diagram, input.getNewText());
-                    return new EventHandlerResponse(true, representation -> true, new EditLabelSuccessPayload(diagram));
+                    response = new EventHandlerResponse(true, representation -> true, new EditLabelSuccessPayload(diagram));
+                }
+            } else if (input.getLabelId().endsWith(CENTERLABEL_SUFFIX)) {
+                String inputId = this.extractEdgeId(input.getLabelId());
+                var edge = this.diagramService.findEdgeById(diagram, inputId);
```

Comment:
Should be replaced by a call to the method `IDiagramQueryService#findEdgeByLabelId` to create

Reply:
Done